### PR TITLE
doc updates for syms macro

### PR DESCRIPTION
**Pull request by jverzani**

### CODE REVIEW

**Comment by jverzani** [comment]
The doctests may not pass, but I can address that later, as they introduce a lot of noise that would distract from the focus here.

File: src/symfunction.jl [hunk 1]
```diff
@@ -22,7 +22,11 @@ F,G,H = SymFunction("F, G, H")
 
 This is just a thin wrapper around `sympy.Functioni` for symbolic functions that allows prime notation in place of using `diff`.
 
-The macro `@symfuns` is also available for constructing symbolic functions.
+The macro [`@syms`](@ref) is also available for constructing symbolic functions.
+
+```
+@syms u(t), v(t)::real
```
Comment (matthieubulte) [inline src/symfunction.jl]: I think this should be `@syms u(), v()::real`.

**Comment by matthieubulte** [comment]
Wow, this is epic! Is there a way to execute the docs to make sure they all run?

**Comment by jverzani** [comment]
Yes, no `(t)`. Thanks. I also noted an extraneous `i`: ``sympy.Functioni`. I'll fix and push the change.

As for running the docs, there is

```
using Documenter, SymPy
doctest(SymPy, fix=true)
```

I have run that locally, but want to merge this in before merging those changes. (They have a lot of churn due to how vectors are printed differently with v1.6).

**Comment by jverzani** [comment]
I see what you are getting at, I also forgot to make those snippets doctests. Will address that too.

**Comment by jverzani** [comment]
Okay, the doctests pass with this latest 

```
julia> doctest(Polynomials, fix=true)
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: Skipped ExpandTemplates step (doctest only).
[ Info: Skipped CrossReferences step (doctest only).
[ Info: Skipped CheckDocument step (doctest only).
[ Info: Skipped Populate step (doctest only).
[ Info: Skipped RenderDocument step (doctest only).
Test Summary:         | Pass  Total
Doctests: Polynomials |    1      1
Test.DefaultTestSet("Doctests: Polynomials", Any[], 1, false, false)
```

**Comment by matthieubulte** [comment]
@jverzani I added support for tensor declaration in my last commit. Maybe you want to rebase on my master to check it out 😉

**Pull request merged**